### ISSUE TITLE
Ghost variables and interfaces

### ISSUE THREAD

**Issue by adilsonv77** [OpenJML/OpenJML#547]
I have the interface and class below. When I run class C, it gives me an exception:

`Exception in thread "main" java.lang.ClassFormatError: Illegal field modifiers in class I:`

However, it works when I used abstract class instead interface.
```
public interface  I {

	//@ public ghost int state;
}
```
```
public class C implements I {

	public static void main(String[] args) { 
		
		new C();
		
	}
	
}
```

**Comment by davidcok** [comment]
If you have

> public interface  I {
> 
> 	public ghost state;
> }
You will get a compiler error, so OpenJML should emit a compiler error for your example as well.

Both cases work if the field is declared ‘final’ (and initialized).

- DavID

**Comment by adilsonv77** [comment]
It was not `public ghost state`, but `public ghost int state`.

Obviously, I simplified the example. I cannot use final modifier in this case, because it is a changeable variable . I try to initialize the variable and also it didn't work,

**Comment by davidcok** [comment]
Instance variables are not allowed in interfaces - ghost or not. (Model fields that have represents clauses in derived classes are allowed).
Implementing such a feature for RAC is a major task.

**Comment by leavens** [comment]
Hi David,

JML has always allowed model instance fields in interfaces.



        Regards,

        Gary T. Leavens
        437D Harris Center (Bldg. 116)
        Computer Science, University of Central Florida
        4000 Central Florida Blvd., Orlando, FL 32816-2362 USA
        http://www.cs.ucf.edu/~leavens  phone: +1-407-823-4758
        leavens@cs.ucf.edu

**Comment by davidcok** [comment]
Yes - I was not precise enough. Model instance variables are implemented for both esc and rac in OpenJML.

**Comment by davidcok** [comment]
Tests and errors and warnings cleaned up for V21-0.18
